Formula/gotify-cli.rb

```diff
@@ -0,0 +1,25 @@
+class GotifyCli < Formula
+  desc "Command-line interface for pushing messages to gotify/server"
+  homepage "https://github.com/gotify/cli"
+  url "https://github.com/gotify/cli/archive/refs/tags/v2.2.0.tar.gz"
+  sha256 "d4e6fbc087c6cc4b320b3fcae4c28d65d514b7e55c95e4c0e764abb3a6c94064"
+  license "MIT"
+  head "https://github.com/gotify/cli.git", branch: "master"
+
+  livecheck do
+    url "https://github.com/gotify/cli/releases/latest"
+    regex(%r{href=.*?/tag/v?(\d+(?:\.\d+)+)["' >]}i)
+  end
+
+  depends_on "go" => :build
+
+  def install
+    system "go", "build", *std_go_args(ldflags: "-X main.Version=#{version}
+                                       -X main.BuildDate=2021-12-09-12:45:29
+                                       -X main.Commit=c8bb424d497f42661e061e93f9e0068852b14975")
+  end
+
+  test do
+    assert_match "Version:   #{version}", shell_output("#{bin}/gotify-cli v | head -n 1")
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Valid point, thanks. I'm trying to write a test, which should fail, but can't seem to find the proper function for it.

`shell_output("#{bin}/gotify-cli p test")` will return with error code 1 (which is ok) and the message `token is not configured, run 'gotify init'`. Which function can I use for this? Right now `assert_match` with message and shell_output fails because the of the child process:

```
An exception occurred within a child process:
  Minitest::Assertion: Expected: 0
  Actual: 1
```

Reply:
`shell_output("#{bin}/gotify-cli p test", 1)` you can add the expected output as the second argument

Reply:
Perfect, thank you!